### PR TITLE
Fast registration: Make download destination dir configurable

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
To support using pyflyte outside of the container, this change adds a `dest-dir` param to fast-registration so code distributions can be installed wherever a user decides.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
https://docs.google.com/document/d/1n7DmKXaOv8dAb9eHD4TKqka6FsdJNs3Tugpf1I6nvMg/edit?usp=sharing

## Tracking Issue
https://github.com/lyft/flyte/issues/554

## Follow-up issue
_NA_

### CODE REVIEW

**Review by kumare3 — APPROVED** [review]
Awesome!

**Review by wild-endeavor — APPROVED** [review]
+1 and docs if you have time plz
